### DISCUSSION TITLE
What is JIT trying to do?

### DISCUSSION THREAD

**Discussion by vish-pr** [tinygrad/tinygrad#2341] [Q&A] [answered]
This is a very basic question, and reading code should clear it, I have tried it and I think I understand it some kind of caching on request and response of functions. But still I am not sure,
1) What is problem statement it is solving?
2) How can we cache responses, if it will be different for different for input?

**Comment by wozeparrot** [comment, marked as answer]
1. Speed
2. It doesn't cache responses, it caches kernels then just runs them without going thru the expensive complication process again.
